filesystem/dir.c

```diff
@@ -0,0 +1,110 @@
+/*
+dir.c - directory operations
+Copyright (C) 2022 Alibek Omarov, Velaron
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+*/
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#if XASH_POSIX
+#include <unistd.h>
+#endif
+#include <errno.h>
+#include <stddef.h>
+#include "port.h"
+#include "filesystem_internal.h"
+#include "crtlib.h"
+#include "xash3d_mathlib.h"
+#include "common/com_strings.h"
+
+void FS_Close_DIR( searchpath_t *search ) {}
+
+void FS_PrintInfo_DIR( searchpath_t *search, char *dst, size_t size )
+{
+	Q_strncpy( dst, search->filename, size );
+}
+
+int FS_FindFile_DIR( searchpath_t *search, const char *path )
+{
+	char netpath[MAX_SYSPATH];
+
+	Q_sprintf( netpath, "%s%s", search->filename, path );
+
+	if( FS_SysFileExists( netpath, !( search->flags & FS_CUSTOM_PATH ) ) )
```

Comment:
What if use [va](https://github.com/FWGS/xash3d-fwgs/blob/master/public/crtlib.c#L627=https://github.com/FWGS/xash3d-fwgs/blob/master/public/crtlib.c#L627=) here?

Reply:
Then it stops working, probably 16 string are not enough